cloud/scope/powervs_cluster.go

```diff
@@ -1129,6 +1169,390 @@ func (s *PowerVSClusterScope) createVPCSubnet(subnet infrav1beta2.Subnet) (*stri
 	return subnetDetails.ID, nil
 }
 
+// ReconcileVPCSecurityGroup reconciles VPC security group.
+func (s *PowerVSClusterScope) ReconcileVPCSecurityGroup() (bool, error) {
+	for _, securityGroup := range s.IBMPowerVSCluster.Spec.VPCSecurityGroups {
+		var securityGroupID *string
+		var securityGroupRuleIDs []*string
+
+		if securityGroup.Name != nil {
+			securityGroupID, securityGroupRuleIDs = s.GetVPCSecurityGroupID(*securityGroup.Name)
+		} else {
+			securityGroupRuleIDs = s.GetVPCSecurityGroupRuleIDs(*securityGroup.ID)
+		}
+
+		if securityGroupID != nil && securityGroupRuleIDs != nil {
+			if _, _, err := s.IBMVPCClient.GetSecurityGroup(&vpcv1.GetSecurityGroupOptions{
+				ID: securityGroupID,
+			}); err != nil {
+				return false, err
+			}
+			for _, rule := range securityGroupRuleIDs {
+				if _, _, err := s.IBMVPCClient.GetSecurityGroupRule(&vpcv1.GetSecurityGroupRuleOptions{
+					SecurityGroupID: securityGroupID,
+					ID:              rule,
+				}); err != nil {
+					return false, err
+				}
+			}
+			continue
+		}
+
+		securityGroupDet, ruleIDs, err := s.validateVPCSecurityGroup(securityGroup)
+		if err != nil {
+			return false, fmt.Errorf("error validating existing security group: %v", err)
+		}
+		if securityGroupDet != nil {
+			s.Logger.Info("Security group already exists", "name", *securityGroupDet.Name)
```

Comment:
I think its ok here, since its going to this block only one time. Same is there on other components too when user provided spec is already exists in cloud, wdys?

Reply:
By default, we are setting the log level to 3 to keep uniformity as per the new changes in [PR](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1755).